src/unitxt/metrics.py

```diff
@@ -74,6 +74,98 @@ class Metric(Artifact):
+            return
+        prediction_type = self.get_prediction_type()
+        if not isoftype(predictions, List[Any]):
+            raise ValueError(
+                f"Metric {self.get_metric_name()} should receive a list of predictions {self.get_metric_name()}.  Received predictions of type {type(predictions)}: {predictions}"
+            )
+
+        if not isoftype(references, List[Any]):
+            raise ValueError(
+                f"Metric {self.get_metric_name()} should receive a list of predictions. Received references of type {type(references)}: {references}"
+            )
+
+        if len(references) != len(predictions):
+            raise ValueError(
+                f"references size ({len(references)})"
+                f" doesn't mach predictions size ({len(references)})."
+            )
+
+        for reference in references:
+            self._validate_reference(prediction_type, reference)
+
+        for prediction in predictions:
+            self._validate_prediction(prediction_type, prediction)
+
+    def _validate_prediction(self, prediction_type, prediction):
+        if prediction is None:
+            if not self.allow_none_prediction:
+                raise ValueError(
+                    f"Prediction is not allowed to be None in {self.get_metric_name()} metric. Received prediction of type {type(prediction)}: {prediction}"
+                )
+        elif not isoftype(prediction, prediction_type):
+            raise ValueError(
+                f"Each prediction is expected to be of type '{self.prediction_type}' in {self.get_metric_name()} metric. Received prediction of type {type(prediction)}: {prediction}"
+            )
+
+    def _validate_reference(self, prediction_type, reference):
+        if not isoftype(reference, List[Any]):
+            raise ValueError(
+                f"Expecting a list of references for each prediction in {self.get_metric_name()} metric. Received reference of type {type(reference)}: {reference}"
+            )
+        if self.single_reference_per_prediction and not len(reference) == 1:
+            raise ValueError(
+                f"Expecting a list with a single reference per prediction in {self.get_metric_name()} metric. Received a list with multiple references: {reference}"
+            )
+        for ref in reference:
+            if not isoftype(ref, prediction_type):
+                raise ValueError(
+                    f"Each reference is expected to be of type '{self.prediction_type}' in {self.get_metric_name()} metric. Received reference of type {type(ref)}: {ref}"
+                )
+
+    def get_prediction_type(self):
+        for type in [
+            "Any",
+            "str",
+            "int",
+            "float",
+            "List[str]",
+            "List[float]",
+            "Dict",
+            "Dict[str,str]",
+            "List[Tuple[str,str]]",
+        ]:
+            if self.prediction_type == type:
+                return eval(type)
```

Comment:
See https://github.com/IBM/unitxt/pull/688 for a more complete solution

Reply:
Looks good. I will use it.  I merged to this PR to test it.  Once the other PR is merged, the changes will disappear from this PR.